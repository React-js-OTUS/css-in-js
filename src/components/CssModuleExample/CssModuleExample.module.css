.root :global(.some-global-class) {
  color: pink;
}

.button {
  border-radius: 8px;
  font-size: 18px;
  padding: 8px 16px;
  outline: none;
  border: 0;
  background-color: #0A67A3;
  color: #fff;
  cursor: pointer;
}

.button:hover {
  background-color: #3E97D1;
}

.button:active {
  background-color: #24587A;
}

.text {
  font-size: 18px;
}

.opened {
  display: block;
}

.closed {
  display: none;
}
