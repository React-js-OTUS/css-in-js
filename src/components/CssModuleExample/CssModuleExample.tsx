import * as React from 'react';
import { FC, useState } from 'react';
import cn from 'clsx';
import s from './CssModuleExample.module.css';

export type ButtonProps = React.HTMLAttributes<HTMLButtonElement> & {
  className?: string;
};

export const CssModuleExample: FC<ButtonProps> = ({ className, ...props }) => {
  const [opened, setOpened] = useState<boolean>(false);
  return (
    <div className={cn(s.root, className)}>
      <button className={s.button} {...props} onClick={() => setOpened(v => !v)} />
      <div className={cn(s.text, opened ? s.opened : s.closed)}>Test</div>
      <div className={cn(s.text, !opened && s.closed)}>Test</div>
      <div className="some-global-class">Global</div>
    </div>
  );
}
