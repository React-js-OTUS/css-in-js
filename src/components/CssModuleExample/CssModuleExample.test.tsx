import React from 'react';
import { render } from '@testing-library/react';
import { CssModuleExample } from './CssModuleExample';

describe('CssModuleButton', () => {
  it('base snapshot', () => {
    const { container } = render(<CssModuleExample>CssModuleButton</CssModuleExample>);
    expect(container).toMatchSnapshot();
  });
});
