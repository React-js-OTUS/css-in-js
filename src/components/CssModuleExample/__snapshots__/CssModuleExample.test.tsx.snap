// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CssModuleButton base snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <button
      class="button"
    >
      CssModuleButton
    </button>
    <div
      class="text closed"
    >
      Test
    </div>
    <div
      class="text closed"
    >
      Test
    </div>
    <div
      class="some-global-class"
    >
      Global
    </div>
  </div>
</div>
`;
